src/status_im2/contexts/onboarding/common/carousel/view.cljs

```diff
@@ -0,0 +1,96 @@
+(ns status-im2.contexts.onboarding.common.carousel.view
+  (:require [quo2.core :as quo]
+            [utils.i18n :as i18n]
+            [utils.re-frame :as rf]
+            [react-native.core :as rn]
+            [react-native.navigation :as navigation]
+            [react-native.reanimated :as reanimated]
+            [status-im2.common.resources :as resources]
+            [status-im2.contexts.onboarding.common.carousel.style :as style]
+            [status-im2.contexts.onboarding.common.carousel.animation :as animation]))
+
+(def header-text
+  [{:text     (i18n/label :t/join-decentralised-communities)
+    :sub-text (i18n/label :t/participate-in-the-metaverse)}
+   {:text     (i18n/label :t/chat-with-friends)
+    :sub-text (i18n/label :t/with-full-encryption)}
+   {:text     (i18n/label :t/own-your-crypto)
+    :sub-text (i18n/label :t/use-the-multichain-wallet)}
+   {:text     (i18n/label :t/discover-web3)
+    :sub-text (i18n/label :t/explore-the-decentralized-web)}])
+
+(defn header-text-view
+  [index window-width]
+  [rn/view {:style (style/header-text-view window-width)}
+   [quo/text
+    {:style  style/carousel-text
+     :weight :semi-bold
+     :size   :heading-2}
+    (get-in header-text [index :text])]
+   [quo/text
+    {:style style/carousel-sub-text
+     :size  :paragraph-1}
+    (get-in header-text [index :sub-text])]])
+
+(defn content-view
+  [window-width status-bar-height index]
+  (let [content-width (* 4 window-width)]
+    [:<>
+     [rn/image
+      {:style  (style/background-image (* 4 window-width))
+       :source (resources/get-image :onboarding-illustration)}]
+     [rn/view {:style (style/header-container status-bar-height content-width index)}
+      [header-text-view 0 window-width]
+      [header-text-view 1 window-width]
+      [header-text-view 2 window-width]
+      [header-text-view 3 window-width]]]))
+
+(defn progress-bar
+  [static? progress-bar-width]
+  [rn/view
+   {:style {:position       :absolute
+            :top            0
+            :width          progress-bar-width
+            :flex-direction :row}}
+   [rn/view {:style (style/progress-bar-item static? false)}]
+   [rn/view {:style (style/progress-bar-item static? false)}]
+   [rn/view {:style (style/progress-bar-item static? false)}]
+   [rn/view {:style (style/progress-bar-item static? true)}]])
+
+(defn dynamic-progress-bar
+  [progress-bar-width]
+  [:f>
+   (fn []
+     (let [width          (animation/dynamic-progress-bar-width progress-bar-width)
+           original-style style/dynamic-progress-bar
+           animated-style (reanimated/apply-animations-to-style
+                           {:width width}
+                           original-style)]
```

Comment:
I'd prefer to call `reanimated/apply-animations-to-style` directly on the style file in `dynamic-progress-bar` and pass it to `:style` prop

Reply:
I don't know, maybe we just want style.cljs files to have styles and avoid animation code. I don't have any strong opinion on this. Let's discuss this with the team.
- Should we move function (apply-animation-to-style/...) function to style.cljs file itself or call where it is used?
- Add conclusion to [new-guidelines.md](https://github.com/status-im/status-mobile/blob/develop/doc/new-guidelines.md)?
- Also make changes in already existing code?

cc: @flexsurfer @ilmotta @J-Son89

Reply:
@OmarBasem you've worked on a lot of animations too, wdyt?

Reply:
I remember I had this same conversation with @briansztamfater some time ago. When I tried to refactor calls to `apply-animation-to-style` outside the `style.cljs` files and into the view files, I didn't really preferred the refactored code, so I ended up agreeing with Brian. I have a slight preference for keeping these function calls on the `style.cljs` file, but any choice makes sense to me.

Reply:
@ilmotta If I recall correctly, I think we agreed to have `apply-animations-to-style` in the `style` file. Is it in the guidelines?

Reply:
moved to style.cljs. Thank you everyone
(we should we add this to new-guidlines, otherwise we will have two versions of it)

Reply:
Thanks for your input everyone, added to our guideline here https://github.com/status-im/status-mobile/pull/15517